src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -154,19 +154,24 @@ class ProcessApiServiceImpl(
     institutionId: Option[String],
     institutionExternalId: Option[String]
   )(bearer: String)(implicit contexts: Seq[(String, String)]): Future[Option[PartyManagementDependency.Institution]] = {
-    institutionId.fold {
-      institutionExternalId.traverse(externalId =>
-        partyManagementService.retrieveInstitutionByExternalId(externalId)(bearer)
-      )
-    } { institutionId =>
-      partyManagementService
-        .retrieveInstitution(UUID.fromString(institutionId))(bearer)
-        .map(institution =>
-          institutionExternalId
-            .flatMap(externalId => Option.when(institution.externalId === externalId)(institution))
-            .orElse(Option(institution))
+    institutionId
+      .fold {
+        institutionExternalId.traverse(externalId =>
+          partyManagementService.retrieveInstitutionByExternalId(externalId)(bearer)
         )
-    }
+      } { institutionId =>
+        partyManagementService
+          .retrieveInstitution(UUID.fromString(institutionId))(bearer)
```

Comment:
`UUID.fromString` throws an exception if the given string is not a uuid

Reply:
the institutionId is supposed to be a UIID:
fixed getInstitutionByOptionIdAndOptionExternalId method firm